core/bin/snapshots_creator/src/main.rs

```diff
@@ -0,0 +1,291 @@
+    let latency = METRICS.storage_logs_processing_duration.start();
+    let mut conn = pool.access_storage_tagged("snapshots_creator").await?;
+    let logs = conn
+        .snapshots_creator_dal()
+        .get_storage_logs_chunk(miniblock_number, hashed_keys_range)
+        .await
+        .context("Error fetching storage logs count")?;
+    drop(conn);
+    let storage_logs_chunk = SnapshotStorageLogsChunk { storage_logs: logs };
+    let key = SnapshotStorageLogsStorageKey {
+        l1_batch_number,
+        chunk_id,
+    };
+    let filename = blob_store
+        .put(key, &storage_logs_chunk)
+        .await
+        .context("Error storing storage logs chunk in blob store")?;
+
+    let output_filepath_prefix = blob_store.get_storage_prefix::<SnapshotStorageLogsChunk>();
+    let output_filepath = format!("{output_filepath_prefix}/{filename}");
+
+    let elapsed = latency.observe();
+    let tasks_left = METRICS.storage_logs_chunks_left_to_process.dec_by(1) - 1;
+    tracing::info!(
+                "Finished chunk number {chunk_id}, overall_progress {}/{}, step took {elapsed:?}, output stored in {output_filepath}",
+                chunks_count - tasks_left,
+                chunks_count
+            );
+
+    Ok(output_filepath)
+}
+
+async fn process_factory_deps(
+    blob_store: &dyn ObjectStore,
+    pool: &ConnectionPool,
+    miniblock_number: MiniblockNumber,
+    l1_batch_number: L1BatchNumber,
+) -> anyhow::Result<String> {
+    let latency = METRICS.factory_deps_processing_duration.start();
+    let mut conn = pool.access_storage_tagged("snapshots_creator").await?;
+    let factory_deps = conn
+        .snapshots_creator_dal()
+        .get_all_factory_deps(miniblock_number)
+        .await?;
+    let factory_deps = SnapshotFactoryDependencies { factory_deps };
+    drop(conn);
+    let filename = blob_store
+        .put(l1_batch_number, &factory_deps)
+        .await
+        .context("Error storing factory deps in blob store")?;
+    let output_filepath_prefix = blob_store.get_storage_prefix::<SnapshotFactoryDependencies>();
+    let output_filepath = format!("{output_filepath_prefix}/{filename}");
+    let elapsed = latency.observe();
+    tracing::info!(
+        "Finished factory dependencies, step took {elapsed:?} , output stored in {}",
+        output_filepath
+    );
+    Ok(output_filepath)
+}
+
+async fn run(
+    blob_store: Box<dyn ObjectStore>,
+    replica_pool: ConnectionPool,
+    master_pool: ConnectionPool,
+) -> anyhow::Result<()> {
+    let latency = METRICS.snapshot_generation_duration.start();
+
+    let config = SnapshotsCreatorConfig::from_env().context("SnapshotsCreatorConfig::from_env")?;
+
+    let mut conn = replica_pool
+        .access_storage_tagged("snapshots_creator")
+        .await?;
+
+    // we subtract 1 so that after restore, EN node has at least one l1 batch to fetch
+    let l1_batch_number = conn.blocks_dal().get_sealed_l1_batch_number().await? - 1;
```

Comment:
This may theoretically panic if there's only the genesis L1 batch in the storage (e.g., in tests).